tests/components/mcp_server/test_http.py

```diff
@@ -0,0 +1,345 @@
+"""Test the Model Context Protocol Server init module."""
+
+from collections.abc import AsyncGenerator
+from contextlib import asynccontextmanager
+from http import HTTPStatus
+import json
+import logging
+
+import mcp
+import mcp.client.session
+import mcp.client.sse
+import pytest
+
+from homeassistant.components.conversation import DOMAIN as CONVERSATION_DOMAIN
+from homeassistant.components.homeassistant.exposed_entities import async_expose_entity
+from homeassistant.components.light import DOMAIN as LIGHT_DOMAIN
+from homeassistant.components.mcp_server.http import MESSAGES_API, SSE_API
+from homeassistant.config_entries import ConfigEntryState
+from homeassistant.const import CONF_LLM_HASS_API, STATE_OFF, STATE_ON
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers import device_registry as dr, entity_registry as er
+from homeassistant.setup import async_setup_component
+
+from tests.common import MockConfigEntry, setup_test_component_platform
+from tests.components.light.common import MockLight
+from tests.typing import ClientSessionGenerator
+
+_LOGGER = logging.getLogger(__name__)
+
+TEST_ENTITY = "light.kitchen"
+INITIALIZE_MESSAGE = {
+    "jsonrpc": "2.0",
+    "id": "request-id-1",
+    "method": "initialize",
+    "params": {
+        "protocolVersion": "1.0",
+        "capabilities": {},
+        "clientInfo": {
+            "name": "test",
+            "version": "1",
+        },
+    },
+}
+EXPECTED_ENDPOINT_EVENT = "event: endpoint\r\n"
+DATA_PREFIX = "data: "
+EXPECTED_ENDPOINT_DATA = f"{DATA_PREFIX}{MESSAGES_API.format(session_id='')}"
+EXPECTED_ENDPOINT_MESSAGE = "event: message\r\n"
+
+
+@pytest.fixture
+async def setup_integration(hass: HomeAssistant, config_entry: MockConfigEntry) -> None:
+    """Set up the config entry."""
+    await hass.config_entries.async_setup(config_entry.entry_id)
+    assert config_entry.state is ConfigEntryState.LOADED
+
+
+@pytest.fixture(autouse=True)
+async def mock_entities(
+    hass: HomeAssistant,
+    device_registry: dr.DeviceRegistry,
+    entity_registry: er.EntityRegistry,
+    setup_integration: None,
+) -> None:
+    """Fixture to expose entities to the conversation agent."""
+    entity = MockLight("kitchen", STATE_OFF)
+    entity.entity_id = TEST_ENTITY
+    setup_test_component_platform(hass, LIGHT_DOMAIN, [entity])
+
+    assert await async_setup_component(
+        hass,
+        LIGHT_DOMAIN,
+        {LIGHT_DOMAIN: [{"platform": "test"}]},
+    )
+
+    async_expose_entity(hass, CONVERSATION_DOMAIN, TEST_ENTITY, True)
```

Comment:
Not related to this PR, but we should come up with a better way to expose entities. We can't tie everything that is not Google/Amazon to conversation.

Reply:
Yeah, i don't really think i understand the product direction here at the moment for this being tied together.

Having Home Assistant as an MCP client also raises some other broader points: Do we add exposed tools to another API? or add them to Assist? Do we want to find a way to make discovered prompts usable with other models? e.g. revisit config sub entries.

It seems like there are more degrees of freedom generally that we'll want here and exposed entities is one example.